File: can_make_pali_queries.dart
class Solution {
  List<bool> canMakePaliQueries(String s, List<List<int>> queries) {
    List<bool> res = [];
    int n = s.length;
    List<int> countList = List.generate(n + 1, (index) => 0);
    for (int i = 0; i < n; i++) {
      countList[i + 1] =
          countList[i] ^ (1 << s[i].codeUnitAt(0) - 'a'.codeUnitAt(0));
    }
    for (List<int> query in queries) {
      int l = query[0], r = query[1], k = query[2];
      int bits = 0;
      int x = countList[r + 1] ^ countList[l];
      while (x > 0) {
        x &= x - 1;
        bits++;
      }
      res.add(bits ~/ 2 <= k);
    }
    return res;
  }
}

File: can_make_pali_queries.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Boolean> canMakePaliQueries(String s, int[][] queries) {
        int n = s.length();
        int[] countList = new int[n + 1];
        for (int i = 0; i < n; i++) {
            countList[i + 1] = countList[i] ^ (1 << s.charAt(i) - 'a');
        }
        List<Boolean> res = new ArrayList<Boolean>();
        for (int[] query : queries) {
            int l = query[0], r = query[1], k = query[2];
            int bits = 0, x = countList[r + 1] ^ countList[l];
            while (x > 0) {
                x &= x - 1;
                bits++;
            }
            res.add(bits <= 2 * k + 1);
        }
        return res;
    }
}